### DISCUSSION TITLE
New ligands docking

### DISCUSSION THREAD

**Discussion by PARODBE** [volkamerlab/teachopencadd#173] [Q&A] [answered]
Hi!

Congrats for this amazing library!. Imagine, that you have created a generative model for ligand generation and you have 30-40 new ligands which aren't include in pdb obviously. Could you provide code or and example doing docking without using pdb? You could some random smiles as an example, is not necessary include the generative model. Also, it would be interesting generate the docking for a set of smiles instead of only one smile.

Thanks!

**Comment by AndreaVolkamer** [comment, marked as answer]
@PARODBE thanks for the positive feedback.
I'm not 100% sure if I understand your question correctly. Setting the generative modelling part aside (for which we do not provide talktorials yet), [T015](https://github.com/volkamerlab/teachopencadd/blob/master/teachopencadd/talktorials/T015_protein_ligand_docking/talktorial.ipynb) and [T018](https://github.com/volkamerlab/teachopencadd/blob/ct-018-pipeline/teachopencadd/talktorials/T018_automated_cadd_pipeline/talktorial.ipynb) (the latter still lives in [PR 125](https://github.com/volkamerlab/teachopencadd/pull/125), but will be merged soon) exemplify how to dock molecules starting from a SMILES string only.  T015 shows the process for a single molecule from SMILES (see `smiles_to_pdbqt` function), T018 contains functions to process a set of molecules represented as SMILES (see `__init__`in [docking class](https://github.com/volkamerlab/teachopencadd/blob/ct-018-pipeline/teachopencadd/talktorials/T018_automated_cadd_pipeline/utils/docking.py) using an openbabel-based helper function ` obabel.create_pdbqt_from_smiles`).

**Comment by PARODBE** [comment]
Thanks!, may be another option could be using mda, I have built it in this way:

![image](https://user-images.githubusercontent.com/62376157/141108816-ed618760-0812-417b-b732-1c154cf7a43b.png)

And after we can follow your code of run_smina.

By the way, when you select the pocket size and pocket center: 
pocket_center = (ligand.positions.max(axis=0) + ligand.positions.min(axis=0)) / 2
pocket_size =ligand.positions.max(axis=0) - ligand.positions.min(axis=0) + 5

I understand that the reason of sum 5 to pocket size is for cover more space or possibilities, or may be not? and in pocket_center the division /2??

Thanks for all!

Pablo

**Reply by AndreaVolkamer** [reply]
Thanks for the mda code snippet!

Regarding the pocket center questions:
* the pocket center is positioned half way between the two extrem values of the ligand atoms' coordinates, e.g. the max-min/2
* and yes, the pocket size is enlarged by 5 to be able to sample the space around the bound ligand more exhaustively (5A or sth in that range is often chosen, though still an arbitrary number)
